### PR TITLE
Profile Page and Masonry Grid

### PR DESCRIPTION
**Pull request by Vladmidir**
# Summary
- Implemented the profile page. 
- Fixed the Editor bugs. The diagrams up to 50mb can now be stored. Also fixed other minor editor bugs.

# Screenshots
image
image
image
For the user's own profile, they can see the challenges for which they started working on a diagram. 
image

### CODE REVIEW

**Comment by o-bm** [comment]
really like what you did with the home page. looks good.

**Comment by o-bm** [comment]
where is the diagrams being stored? im concerned on allowing 50mb per diagram for every challenge for every user is a bit crazy ngl..

File: server/index.js [hunk 1]
```diff
@@ -11,7 +11,8 @@ const checkRole = require("./middleware/CheckRoleMiddleware");
 
 const app = express();
 
-app.use(express.json());
+app.use(express.json({limit: '50mb'}));
+app.use(express.urlencoded({limit: '50mb'}));
```
Comment (o-bm) [inline server/index.js:15]: this

**Comment by o-bm** [comment]
overall looks good

File: client/scss/custom.scss
Comment (erensunerr) [inline client/scss/custom.scss:1]: Great work on doings things the bootstrap-y way.

File: server/models/SolutionInProgress.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ module.exports = (sequelize, DataTypes) => {
         allowNull: false,
       },
       diagram: {
-        type: DataTypes.STRING,
+        type: "TEXT",
```
Comment (erensunerr) [inline server/models/SolutionInProgress.js:22]: You can make default "", which is different from null.

File: server/routes/SolutionInProgressRoutes.js [hunk 3]
```diff
@@ -1,6 +1,10 @@
 const SolutionInProgress = require('../controllers/SolutionInProgressController');
 const router = require("express").Router();
 
+// Get all solutions in progress from the database for the currently logged in user.
+router.get("/myinprogress", SolutionInProgress.findMyInProgress);
+
+// DEVELOPMENT ONLY! Get all solutions in progress from the database. 
 router.get("/", SolutionInProgress.findAll);
```
Comment (erensunerr) [inline server/routes/SolutionInProgressRoutes.js:8]: Consider wrapping this in `process.env?.ENV == "dev"`

**Review by erensunerr — APPROVED** [review]

**Review by arthurhohinng — APPROVED** [review]
LGTM

**Pull request merged**